File: php/src/Advertisements/Advertisement/Domain/ValueObjects/AdvertisementStatus.php
<?php
declare(strict_types=1);

namespace Demo\App\Advertisements\Advertisement\Domain\ValueObjects;

enum AdvertisementStatus
{
    case ENABLED;
    case DISABLED;

    public static function fromString(string $role): self
    {
        return match ($role) {
            'enabled' => self::ENABLED,
            'disabled' => self::DISABLED,
        };
    }

    public function value(): string
    {
        return match ($this) {
            self::ENABLED => 'enabled',
            self::DISABLED => 'disabled',
        };
    }
}

File: kotlin/src/main/kotlin/advertisements/advertisement/domain/value_object/AdvertisementStatus.kt
package advertisements.advertisement.domain.value_object

enum class AdvertisementStatus {
    ENABLED,
    DISABLED;

    companion object {
        fun fromString(role: String): AdvertisementStatus {
            return when (role) {
                "enabled" -> ENABLED
                "disabled" -> DISABLED
                else -> throw IllegalArgumentException("Unknown role: $role")
            }
        }
    }

    fun value(): String {
        return when (this) {
            ENABLED -> "enabled"
            DISABLED -> "disabled"
        }
    }
}
